pageserver/src/walingest.rs

```diff
@@ -304,8 +304,11 @@ impl<'a, R: Repository> WalIngest<'a, R> {
                 image.resize(image.len() + blk.hole_length as usize, 0u8);
                 image.unsplit(tail);
             }
-            image[0..4].copy_from_slice(&((lsn.0 >> 32) as u32).to_le_bytes());
-            image[4..8].copy_from_slice(&(lsn.0 as u32).to_le_bytes());
+            if image[10] != 0 || image[11] != 0 {
```

Comment:
Hmm, this assumes that the page uses the "standard page layout", i.e PageHeaderData. That's not guaranteed. However, I see that XLogReadBufferForRedoExtended() makes the same assumption. It doesn't seem safe in upstream either...

Reply:
Let's add a PageIsNew() function, to make it clear what this does, at least, and add a comment explaining that this matches the behaviour of XLogReadBufferForRedoExtended()

Reply:
Okay, good point on the checksums..